api/control/v1alpha1/types.go

```diff
@@ -202,66 +202,33 @@ type BindingSpec struct {
 }
 
 // DownsyncObjectReferences explicitly defines the objects to be down-synced.
-// The ClusterScope list defines the cluster-scope objects, NamespaceScope list
-// defines the namespace-scope objects and WorkloadGeneration represents the
-// generation of the objects in the ClusterScope and NamespaceScope lists.
-// Upon a change in any of workload objects that should be distributed
-// (e.g., spec, annotations or labels) the workload generation field should be incremented.
+// The ClusterScope list defines the cluster-scope objects and NamespaceScope list
+// defines the namespace-scope objects.
 type DownsyncObjectReferences struct {
-	// `clusterScope` holds a list of individual cluster-scoped objects
-	// to downsync, organized by resource.
-	// Remember that a "resource" is a kind/type/sort of objects,
-	// not an individual object.
+	// `clusterScope` holds a list of individual cluster-scoped objects to downsync.
```

Comment:
By including the object's ResourceVersion, the semantics here is not simply a set of object references. I think you do not intend this to be a set of (object reference, ResourceVersion) pairs; that would allow the set to have two entries that have the same object reference but different ResoureVersion.

I think the intended semantic here is both a set of object references and a ResourceVersion per object reference. In other words, this is logically an enumerated map from object reference to ResourceVersion.

It is necessary to clearly say the intended semantics in English. Further, I recommend also saying the intended semantics in the usual ways for Kubernetes API object type definitions. For an example, look at PodSpec in https://github.com/kubernetes/api/blob/v0.28.2/core/v1/types.go for how the map from volume name to VolumeSource is handled. FYI, the patchMergeKey can be a comma-separated-list of field names. There is a definition of additional markup in https://v1-28.docs.kubernetes.io/docs/reference/using-api/server-side-apply/#merge-strategy ; I do not understand why it is not there in core/v1/types.go.